doc/source/container_builder.rst

```diff
@@ -0,0 +1,282 @@
+Building in a Self-Contained Environment
+========================================
+
+.. topic:: Abstract
+
+   Users building images with KIWI face problems if they want
+   to build as non root user or can't install KIWI on their
+   system due to inappropriate build host requirements or
+   project policies.
+
+   This document describes how to perform the build process in
+   a Docker container using the Dice containment build system
+   written for KIWI.
+
+   The changes to the machine to become a build host will
+   be reduced to the requirements of Dice and Docker.
+
+Requirements
+------------
+
+The following components needs to be installed on the build system:
+
+* Dice - a containment build system for KIWI.
+
+* Docker - a container framework based on the Linux
+  container support in the kernel.
+
+* Docker Build Container for KIWI.
+
+* optionally Vagrant - a framework to run, provision and control
+  virtual machines and container instances. Vagrant has a very nice
+  interface to provision a machine prior to running the actual build.
+  It also supports docker as a provider which makes it a perfect fit
+  for complex provisioning tasks in combination with the Docker
+  container system.
+
+Installing and Setting up Dice
+------------------------------
+
+The Dice packages and sources are available at the following locations:
+
+* Build service project: https://build.opensuse.org/project/show/Virtualization:Appliances:ContainerBuilder
+* Sources: https://github.com/SUSE/dice
+
+.. code:: bash
+
+    $ sudo zypper in ruby2.1-rubygem-dice
+
+Installing and Setting up Docker
+--------------------------------
+
+Docker packages are usually available with the used distribution.
+
+.. code:: bash
+
+    $ sudo zypper in docker
```

Comment:
Should we distinguish here between the different distributions? Probably they are named the same (`docker`?) on all systems?